pypeit/core/datacube.py

```diff
@@ -542,27 +545,39 @@ def compute_weights(all_ra, all_dec, all_wave, all_sci, all_ivar, all_idx, white
     return all_wghts
 
 
-def coadd_cube(files, parset, overwrite=False):
+def coadd_cube(files, spectrograph=None, parset=None, overwrite=False):
```

Comment:
The dependence on spectrograph is not necessary. It can be instantiated from the information in the file header. We adopt this model in all post-processing codes and I would appreciate it that you not deviate from that convention here.

Reply:
The issue is that I now need the detector name before loading the `Spec2DObj`.  That's easiest using a `Spectrograph` instance.  The default behavior is to pull the spectrograph name from the spec2D files, but that means I effectively need to open the file twice.  All I've done is introduce the capability of the calling function to provide the spectrograph directly if it already knows what it is.  So I haven't introduced a new dependency here, all I've done is enable optional ways to instantiate spectrograph, which was in the previous function.  Do you want me to remove `spectrograph` as an optional argument?

Reply:
I don't feel strongly about it. But my suggestion is that remove spectrograph as an input optional argument and simply read the header. Then you you can just use fits.getheader() which may be faster than opening the file. Even if it is not faster,  I think the milliseconds it takes to open the file such that you can read the header is better than a more complicated argument list, more complicated docs to accommodate a more complex argument list,  and an extra 15 lines of code.